controllers/gateway_controller_reconciler_utils.go

```
@@ -71,58 +71,53 @@ func (r *GatewayReconciler) createControlPlane(
 	return r.Client.Create(ctx, controlplane)
 }
 
-func (r *GatewayReconciler) ensureGatewayMarkedReady(ctx context.Context, gateway *gatewayDecorator, dataplane *operatorv1alpha1.DataPlane) error {
-	if !k8sutils.IsReady(gateway) {
-		services, err := k8sutils.ListServicesForOwner(
-			ctx,
-			r.Client,
-			consts.GatewayOperatorControlledLabel,
-			consts.DataPlaneManagedLabelValue,
-			dataplane.Namespace,
-			dataplane.UID,
-		)
-		if err != nil {
-			return err
-		}
-
-		count := len(services)
-		if count > 1 {
-			return fmt.Errorf("found %d services for DataPlane currently unsupported: expected 1 or less", count)
-		}
+func (r *GatewayReconciler) ensureGatewayConnectivityStatus(ctx context.Context, gateway *gatewayDecorator, dataplane *operatorv1alpha1.DataPlane) (message string) {
```

Why do we want to change the function signature to return `string` here? I know the purpose here is to directly use the returned value as message in the condition, but IMO it is better to return an error here to annotate that something incorrect happened.

---

No specific reason. Since we'd use that error as a string, why not return a string? Just a matter of readability, but I can change the signature and return an error if you deem it :)